backend/msgpack/writer.go

```diff
@@ -142,11 +147,7 @@ func (w *writer) Close() error {
 		return errInstanceWriterClosed
 	}
 	w.closed = true
-	if err := w.flushWithLock(); err != nil {
-		w.log.WithFields(
-			xlog.NewLogErrField(err),
-		).Error("flush error when closing writer")
-	}
+	w.flushWithLock() // nolint: errcheck
```

Comment:
Since we're calling `flushWithLock` in a few different places I thought I would put it in the function so we don't have to repeat emitting the metric at each call site. I don't feel strongly about that though, will update.